### PR TITLE
[notifications] mark Subscription type as deprecated

### PR DESCRIPTION
**Pull request by vonovak**
# Why

Marks the export of `Subscription` in expo-notifications as deprecated, people should use `EventSubscription`.

closes https://github.com/expo/expo/pull/33242

# How

- export both `EventSubscription` and `Subscription`, add deprecation
- rebuild docs

# Test Plan

Not needed

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by vonovak** [comment]
* **#33295** <a href="https://app.graphite.dev/github/pr/expo/expo/33295?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/expo/expo/33295?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/expo-notifications/src/Notifications.types.ts [hunk 1]
```diff
@@ -768,6 +768,10 @@ export type MaybeNotificationResponse = NotificationResponse | null | undefined;
 export {
   PermissionExpiration,
   PermissionResponse,
+  /**
+   * @deprecated use the `Subscription` type instead
+   * */
   EventSubscription,
+  EventSubscription as Subscription,
```
Comment (tsapeta) [inline packages/expo-notifications/src/Notifications.types.ts:775]: To be honest I would do the opposite – deprecate `Subscription` in favor of `EventSubscription` which is used elsewhere. In the long term we should export this type from the `expo` package.

File: packages/expo-notifications/src/Notifications.types.ts [hunk 2]
```diff
@@ -768,6 +768,10 @@ export type MaybeNotificationResponse = NotificationResponse | null | undefined;
 export {
   PermissionExpiration,
   PermissionResponse,
+  /**
+   * @deprecated use the `Subscription` type instead
+   * */
```
Comment (tsapeta) [inline packages/expo-notifications/src/Notifications.types.ts:773]: I wouldn't be surprised if TS doesn't recognize this doc comment (inside `export {}`). Have you checked if this appears in the docs?

Comment (vonovak) [reply packages/expo-notifications/src/Notifications.types.ts]: it now renders as expected

**Review by tsapeta — APPROVED** [review]

**Pull request merged**